cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/Extensions.kt

```diff
@@ -612,6 +612,27 @@ val Node?.assigns: List<AssignExpression>
 val Node?.problems: List<ProblemNode>
     get() = this.allChildren()
 
+/**
+ * Returns all [Node] children in this graph, where the [Node.additionalProblems] field is not
+ * empty.
+ */
+val Node?.additionalProblemsNotEmpty: List<Node>
+    get() = this.allChildren<Node> { node -> node.additionalProblems.isNotEmpty() }
+
+/**
+ * Returns all [ProblemNode] children - stored in a [Node.additionalProblems] field - in this graph,
+ * starting with this [Node].
+ */
+val Node?.additionalProblemsUnpacked: List<ProblemNode>
```

Comment:
allProblems?

Reply:
ah. its only “all additional problems”

Reply:
Yes. And I made it "private", as I don't see an external use case.